com.xrtk.spatial-persistence 0.3.0

SimonDarksideJ opened the pull request:
# XRTK - Mixed Reality Toolkit Pull Request

## Overview
<!-- Please provide a clear and concise description of the pull request. -->

Tested implementation ready for release

StephenHodgson committed:
initial project setup

StephenHodgson committed:
updated project build settings (#1)

* updated project build settings

* updated build platforms

* fix min android version

* removed vulkan render api

* fixed bundle version

StephenHodgson committed:
renamed project from com.xrtk.anchors to com.xrtk.spatial-persistence (#3)

* renamed project

* fixed line endings

* renamed a few lingering artifacts

SimonDarksideJ committed:
Initial implementation of the Spatial Persistence management service

Individual providers will have their own service dependencies

Removed ARFoundation dependency as it is not required for the base implementation

SimonDarksideJ committed:
Updated Data Provider registration.

SimonDarksideJ committed:
Updated Service to become a System Implementation instead of an Extension

SimonDarksideJ committed:
Including feedback from discussions

SimonDarksideJ committed:
Updated Interface

SimonDarksideJ committed:
Merge pull request #5 from XRTK/feature/initialImplementation

Feature/initial implementation

SimonDarksideJ committed:
Merge pull request #7 from XRTK/feature/InitialImplementation

Feature/initial implementation

SimonDarksideJ committed:
Updated implementation based on testing

StephenHodgson committed:
Misc cleanup and change requests (#8)

# XRTK - Mixed Reality Toolkit Pull Request

## Overview
<!-- Please provide a clear and concise description of the pull request. -->
cleaned up and improved implementation a bit
fixed a bunch of spelling errors

StephenHodgson reviewed (changes requested):
See change request https://github.com/XRTK/SpatialPersistence/pull/6

StephenHodgson committed:
made the event callbacks private

StephenHodgson committed:
added awaitable async methods for anchor creation

StephenHodgson committed:
changed return types for create anchor (now returns a guid

changed return types for find anchors (returns void)

StephenHodgson committed:
added delete anchors api

StephenHodgson committed:
don't really need to asset for these

SimonDarksideJ committed:
Tweaked docs and added find anchors async method

StephenHodgson committed:
updated packages

StephenHodgson reviewed (changes requested):
Requires https://github.com/XRTK/XRTK-Core/pull/862 release

StephenHodgson committed:
updated project to 2020.3.17f1 (#10)

* updated project to 2020.3.17f1

* disable android build since this package should be xplatform

SimonDarksideJ committed:
Added a new event to handle anchors with invalid data that a client would need to handle/delete/resolve

SimonDarksideJ committed:
Merge pull request #11 from XRTK/fix/invalidanchorprotection

fix: Updated Spatial Persistence interface to report on Failed Anchors that were requested.

StephenHodgson commented on XRTK.SpatialPersistence/Packages/com.xrtk.spatial-persistence/Runtime/Interfaces/IMixedRealitySpatialPersistenceDataProvider.cs:
```diff
@@ -40,7 +40,7 @@ public interface IMixedRealitySpatialPersistenceDataProvider : IMixedRealityData
 
         /// <summary>
-        /// Create a cloud Anchor at a specified location
+        /// Create a cloud Anchor at a specified location async
```
does adding async to the docs really make that big of a difference if the name of the method already has asycn in it?

SimonDarksideJ committed:
nit: comments

StephenHodgson committed:
updated package info

StephenHodgson committed:
GitHub Actions Workflow (#12)

# XRTK - Mixed Reality Toolkit Pull Request

## Overview
Migrate from Azure DevOps to GitHub Actions Workflow

XRTK-Build-Bot committed:
[skip ci] 0.3.0-preview.13

SimonDarksideJ committed:
Feature/docsupdate (#13)

* Updated documentation for creating anchors.

* Updated Data Provider docs

XRTK-Build-Bot committed:
[skip ci] 0.3.0-preview.14

StephenHodgson committed:
fix typos

XRTK-Build-Bot committed:
[skip ci] 0.3.0-preview.15
